server/src/core/user/validator.ts

```diff
@@ -0,0 +1,46 @@
+import { UserCreateArgs, UserNominateBeneficiaryArgs, UserLoginArgs } from './types'
+import { createValidationError } from '../error';
+import { Schema } from '.';
+
+export const validatesCreate = (args: UserCreateArgs) => {
+  const { error } = Schema.userCreateSchema.validate(args);
+  if (error) throw createValidationError(error.details[0].message);
```

Comment:
Since this pattern is repeated from most functions, I think you could abstract it. e.g. 

```c#
export const validatesCreate = createValidatorFromJoiSchema<UserCreateArgs>(schemas.userCreateSchema);
```
 But this can be refactored in a future PR.